compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -737,6 +737,22 @@ object Trees {
     type ThisTree[+T <: Untyped] = Splice[T]
   }
 
+  /** A tree representing a pattern splice `${ pattern }`, `$ident` or `$ident(args*)` in a quote pattern.
+   *
+   *  Parser will only create `${ pattern }` and `$ident`, hence they will not have args.
+   *  While typing, the `$ident(args*)` the args are identified and desugared into a `SplicePattern`
+   *  containing them.
```

Comment:
Why not have the parser take care of this since it's based purely on syntactic information?

Reply:
In the next PR I will introduce `QuotePattern`. At that point `SplicePattern`s will be kept after parser and until we encode the pattern into `unapply` and `'{..}`s.